nexus/db-queries/src/db/datastore/deployment.rs

```diff
@@ -1101,19 +1107,23 @@ impl DataStore {
             nclickhouse_cluster_configs,
             nclickhouse_keepers,
             nclickhouse_servers,
-        ) = conn
-            .transaction_async(|conn| async move {
+        ) = self.transaction_retry_wrapper("blueprint_delete")
```

Comment:
As we add more of these named transaction wrappers, it might make sense to ensure the names are unique somehow. I'm not sure the best way, but maybe making them all consts in one submodule is enough. Thoughts?

Reply:
We may also want to make it easy (or required) to use the name of the enclosing function as the transaction wrapper. I don't think there's a stable way to get that today, though there are a few hack-ish crates that seem to  work OK.

Reply:
I'm split on this. I agree uniqueness is a nice property to have, but it's a little more friction when writing these operations. Maybe that doesn't matter?

Either way -- I'm gonna punt on this from this PR, because I'd need to change pre-existing retry wrappers too, not just these new ones.

Reply:
> I'm split on this. I agree uniqueness is a nice property to have, but it's a little more friction when writing these operations. Maybe that doesn't matter?

The only reason to care is that using the same name would merge potentially unrelated data in the resulting timeseries. But deferring seems fine, just something to keep in mind.